logging/google/cloud/logging/handlers/GAEHandler.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Logging handler for App Engine Flexible
+
+Logs to the well-known file that the fluentd sidecar container on App Engine
+Flexible is configured to read from and send to Stackdriver Logging.
+
+This file is largely copied from:
+
+https://github.com/GoogleCloudPlatform/python-compat-runtime/blob/master/appengine-vmruntime/vmruntime/cloud_logging.py
+
+"""
+
+import json
+import logging.handlers
+import math
+import os
+
+LOG_PATH_TEMPLATE = '/var/log/app_engine/app.{pid}.json'
+MAX_LOG_BYTES = 128 * 1024 * 1024
+LOG_FILE_COUNT = 3
+
+
+class GAEHandler(logging.handlers.RotatingFileHandler):
+    """A handler that writes to the GAE fluentd Stackdriver logging file.
+
+    Writes to the file that the fluentd agent on App Engine Flexbile is
```

Comment:
typo `Flexbile`, but also you should refer to it as `the App Engine flexible environment`. (flexible environment is *not* capitalized).